addon/-private/system/relationships/has-many.js

```diff
@@ -164,8 +164,11 @@ export const HasManyMixin = Ember.Mixin.create({
     //We need to notifyPropertyChange in the adding case because we need to make sure
     //we fetch the newly added record in case it is unloaded
     //TODO(Igor): Consider whether we could do this only if the record state is unloaded
-
     //Goes away once hasMany is double promisified
-    this.notifyPropertyChange(key);
+    //
+    let cache = Ember.meta(this).readableCache();
+    if (typeof cache === 'object' && cache !== null && cache[key] !== undefined) {
```

Comment:
This is just a quick experiment, don't mind the lolcode it is not currently my intention to merge this as is.